synapse/lib/cell.py

```diff
@@ -39,23 +39,29 @@
 '''
 Base classes for the synapse "cell" microservice architecture.
 '''
-def adminapi(f):
 
-    @functools.wraps(f)
-    def func(*args, **kwargs):
+def adminapi(log=True, *args, **kwargs):
```

Comment:
I would prefer not to parameterize adminapi, but rather add a logger call to the *very* few APIs we want to actually log calls for.

Reply:
On the fence here. It isn't *very few*. I counted 20 places that should be logged, out of 34, so more than half. It wouldn't be awful to put that in the decorator function.

Reply:
per discussion - leave it as a function; remove the args/kwargs; add log=false as the the default.